accepted/2020/net5/net5.md

```diff
@@ -0,0 +1,502 @@
+can be explained (and was properly motivated) it created a massive tax: the
+concept count. If you're new to .NET, where would you start? What is the latest
+stack? You may say, “of course that's .NET Core” but how would anyone know that
+by just looking at the names?
+
+We've simplified the world with .NET Standard, in that class library authors
+don't have to think about all the different “boxes” that represent different
+implementations of .NET. It did that by unifying the *API surface* of the
+various .NET implementations. Ironically, this resulted in us having to add yet
+another box, namely .NET Standard.
+
+To make the future saner, we must reduce the number of boxes. We don't want to
+make .NET less flexible, but we want to reduce nonsensical differences that
+purely resulted from us not being open source early enough. For example,
+Mono/Xamarin/Unity are based on a different set of runtimes and frameworks than
+the .NET Framework/Silverlight/UWP/.NET Core lineage. With .NET Standard, we
+have started to remove the differences in the API surface. With .NET 5, the goal
+is to converge these lineages onto a single product stack, thus unifying
+their *implementations*.
+
+![](pic01.png)
+
+While we strive to provide an experience where you don't have to reason about
+the different kinds of .NET, we still don't want to fully abstract away the
+underlying OS, so you'll continue to be able to call OS specific APIs, be that
+via P/Invokes, WinRT, or the Xamarin bindings for iOS and Android APIs.
+
+Now think about developers who start on this stack and can write any application
+for any of the platforms that .NET provides support for. The branding we
+currently have makes no sense to them. To find documentation and tutorials, the
+only two things a developer should need to know is the name and version of their
+technology stack.
+
+Let's contrast this with some of the NuGet packages that developers have to
+author for today's world:
+
+![](pic02.png)
+
+There are a lot of names and version numbers. Knowing who is compatible with who
+is impossible with a decoder ring. We've simplified this greatly with .NET
+Standard, but this still requires a table that maps .NET Standard versions to
+.NET implementation versions.
+
+The proposal is to reuse the existing net TFM and model OS-specific APIs on top
+via a new syntax:
+
+* `net5.0`. This TFM is for code that runs everywhere. It combines and replaces
+  the `netcoreapp` and `netstandard` names. This TFM will generally only include
+  technologies that work cross-platform (modulo pragmatic concessions, like we
+  already did in .NET Standard).
+
+* `net5.0-win`, `net5.0-ios`, `net5.0-android`. These TFMs represent OS specific
+  flavors of .NET 5 that include `net5.0` plus OS-specific bindings.
+
+NuGet should use this next syntax to automatically understand that `net5.0` can
+be consumed from net6-win (but not the other way around). But more importantly,
```

Comment:
```suggestion
be consumed from `net6-win` (but not the other way around). But more importantly,
```

Reply:
Is this supportd to be `net6-win`? Or `net5-win` (really `net5.0-win`)? If it's correct, I suggest adding both so it's less confusing.

Reply:
It's supposed to be `net6.0-win`. Fixed.